api/src/test/java/org/openmrs/module/sync/SyncProgramAttributeTest.java

```diff
@@ -50,6 +52,81 @@ public String getInitialDataset() {
             throw new RuntimeException(e);
         }
 	}
+
+	@Test
+	@Transactional(propagation = Propagation.NOT_SUPPORTED)
+	public void shouldSyncProgramAttributeOfTypeLocation() throws Exception {
+		runSyncTest(new SyncTestHelper() {
+
+			public void runOnChild() throws Exception {
+
+				ProgramAttributeType programAttributeType = programService.getProgramAttributeType(2); // Health Center Location
+				Assertions.assertNotNull(programAttributeType);
+				PatientProgram pp = programService.getPatientProgram(1);
+				Assertions.assertNotNull(pp);
+				Location healthCenterLocation = Context.getLocationService().getLocation("Someplace");
+				Assertions.assertNotNull(healthCenterLocation);
+				PatientProgramAttribute attr = new PatientProgramAttribute();
+				attr.setAttributeType(programAttributeType);
+				attr.setValueReferenceInternal(healthCenterLocation.getUuid());
```

Comment:
You should probably instead do `attr.setValue(healthCenterLocation)`, as this is the way the API would normally be used